core/domain/question_domain.py

```diff
@@ -229,3 +237,64 @@ def update_question_data(self, question_data):
             question_data: dict. A dict representing the question data.
         """
         self.question_data = question_data
+
+    def add_skill(self, skill_name, user_id):
+        """Adds the question id to the question list of the appropriate skill.
+
+        Args:
+            skill_name: str. The name of the skill.
+            user_id: str. The id of the user.
+        """
+        collection_services.update_collection(
+            user_id, self.collection_id, [{
+                'cmd': collection_domain.CMD_ADD_QUESTION_ID_TO_SKILL,
+                'skill_name': skill_name,
+                'question_id': self.question_id
+            }], 'Add a question_id to skill')
+
+    def remove_skill(self, skill_id, user_id):
+        """Removes the question id from the question list of the appropriate
+        skill.
+
+        Args:
+            skill_id: str. The id of the skill.
+            user_id: str. The id of the user.
+        """
+        collection_services.update_collection(
+            user_id, self.collection_id, [{
+                'cmd': collection_domain.CMD_REMOVE_QUESTION_ID_FROM_SKILL,
+                'skill_id': skill_id,
+                'question_id': self.question_id
+            }], 'Remove a question_id from skill')
+
+    def get_skills(self):
+        """Fetches the skills associated with the question."""
+        collection = collection_services.get_collection_by_id(
+            self.collection_id)
+        skills = collection.skills
+
+        question_skills = []
+        for skill_id in skills:
+            if self.question_id in collection.skills[skill_id].question_ids:
+                question_skills.append(collection.skills[skill_id])
+        return question_skills
+
+    def can_user_answer_question(self, user_id, collection_id):
+        """Verifies if the given user has the acquired
+        skills to answer the question.
+
+        Args:
+            user_id: str. The id of the user.
+            collection_id: str. The id of the collection.
+
+        Returns:
+            bool. A boolean representing True or False.
+        """
+        question_skills = self.get_skills()
```

Comment:
You know -- I'm not sure this is particularly efficient, or if you even need the get_skills() method. Why not take the acquired skills and just iterate through those instead? This could potentially save the need to search within long lists, especially if the list corresponds to something the student hasn't seen yet.

Furthermore, I don't actually understand where this method is being called from? Is it even needed? I thought in general that we'd know the student's skills, and based on those skills, find questions to ask them. So I'm a bit unsure about why we would be checking this in the first place, but perhaps there is a use case I'm missing.

Reply:
I think this will be used to update the skills associated with question.
Assume question has skill A and skill B, if the creator wants to add skill C to the question. We can show Question already has skills A and B using get_skills().

I believe this was supposed to be used after obtaining a list of questions based on the skill_ids sent to question_batch method but I think I used a better approach instead maybe.

Reply:
OK, there are two choices here. One is to drop this function and other functions that are currently unneeded, and leave them to future PRs once we're clearer about the use case and have defined controllers for those use cases. However, if we want to keep this, then again this can't be in the question object because it refers to collection_services. It needs to go in question_services.